packages/events/src/once/sync.rs

```diff
@@ -0,0 +1,518 @@
+//! Thread-safe one-time events.
+//!
+//! This module provides thread-safe event types that can be shared across threads
+//! and used for cross-thread communication.
+
+use std::sync::Mutex;
+use std::sync::atomic::{AtomicBool, Ordering};
+
+/// A one-time event that can send and receive a value of type `T`.
+///
+/// This is the thread-safe variant that can create sender and receiver endpoints
+/// that can be used across threads. The event itself doesn't need to be shared
+/// across threads - only the sender and receiver do.
+///
+/// The event can only be used once - after obtaining the sender and receiver,
+/// subsequent calls to obtain them will panic (or return [`None`] for the checked variants).
+///
+/// For single-threaded usage, see [`crate::once::LocalEvent`] which has lower overhead.
+///
+/// # Thread Safety
+///
+/// This type requires `T: Send` as values will be sent across thread boundaries.
+/// The sender and receiver implement `Send + Sync` when `T: Send`.
+///
+/// # Example
+///
+/// ```rust
+/// use events::once::Event;
+///
+/// let event = Event::<String>::new();
+/// let (sender, receiver) = event.endpoints();
+///
+/// sender.send("Hello".to_string());
+/// let message = receiver.receive();
+/// assert_eq!(message, "Hello");
+/// ```
+#[derive(Debug)]
+pub struct Event<T>
+where
+    T: Send,
+{
+    used: AtomicBool,
+    channel: Mutex<Option<(oneshot::Sender<T>, oneshot::Receiver<T>)>>,
```

Comment:
Why do we need this Mutex here? As clarified in previous comment, there is no requirement for `Event` to be thread-safe, only the Sender/Receiver have to be thread-safe, so this feels excessive. If there is a reason, let's document this justification with inline comments. If not, let's simplify.